File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.py
class Solution(object):
    def sumNumbers(self, root):
        if not root:
            return

        def dfs(node, cur_value):
            if not node:
                return 0
            
            cur_value = cur_value * 10 + node.val

            if not node.left and not node.right:
                return cur_value
            
            return dfs(node.left, cur_value) + dfs(node.right, cur_value)
            
            
        return dfs(root, 0)

            
        

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.java
class Solution {
    public int sumNumbers(TreeNode root) {
        return dfs(root, 0);
    }

    private int dfs(TreeNode node, int curSum) {
        if(node == null) return 0;
        curSum = curSum * 10 + node.val;

        if(node.right == null && node.left == null) return curSum;
    

        return dfs(node.left, curSum) + dfs(node.right, curSum);
    }
}